### DISCUSSION TITLE
Scripting: how to save a calculated smart meter value to a variable?

### DISCUSSION THREAD

**Discussion by Kugeleis** [arendst/Tasmota#16125] [Q&A] [answered]
I use in the smart meter section of my script a calculated value `1,=d 1 120 @1,current power,W,Power_curr,16`. 
How do I get this into a variable? Following the [example as here](https://tasmota.github.io/docs/Smart-Meter-Interface/#sanxing-sx6x1-sxxu1x-ascii-obis) I just get the string "1,=d 1 120 @" from
```
>D
p="1,=d 1 120 @"
```

**Comment by gemu2015** [comment, marked as answer]
#define USE_SML_SCRIPT_CMD
enables access to SML decode lines from script
sml[l]   l = decode line (1 .. N)
